torch/csrc/jit/mobile/register_mobile_ops.cpp

```diff
@@ -24,15 +24,132 @@ static auto registry0 = torch::RegisterOperators().op(
     return at::add(a, b, c);
   })
 ).op(
-  "_aten::_convolution",
-  torch::RegisterOperators::options().kernel(c10::TensorTypeId::CPUTensorId,
-  [](at::Tensor input, at::Tensor weight, c10::optional<at::Tensor> bias,
-  std::vector<int64_t> stride, std::vector<int64_t> padding,
-  std::vector<int64_t> dilation, bool transposed, std::vector<int64_t> output_padding,
-  int64_t groups, bool benchmark, bool deterministic, bool cudnn_enabled) {
-  return at::_convolution(input, weight, optional_to_tensor(bias), stride, padding, dilation,
-                     transposed, output_padding, groups, benchmark, deterministic, cudnn_enabled);
-  })
+"_aten::add_.Tensor",
+torch::RegisterOperators::options().kernel(c10::TensorTypeId::CPUTensorId,
+                                           [](at::Tensor a, at::Tensor b, at::Scalar c) ->at::Tensor {
+                                             return at::add(a, b, c);
+                                           })
+).op(
+"_aten::adaptive_avg_pool2d",
+torch::RegisterOperators::options().kernel(c10::TensorTypeId::CPUTensorId,
+                                           [](at::Tensor a, std::vector<int64_t> b) ->at::Tensor {
+                                             return at::adaptive_avg_pool2d(a, b);
+                                           })
+).op(
+"_aten::mm",
+torch::RegisterOperators::options().kernel(c10::TensorTypeId::CPUTensorId,
+                                           [](at::Tensor a, at::Tensor b) ->at::Tensor {
+                                             return at::mm(a, b);
+                                           })
+).op(
+"_aten::_convolution",
+torch::RegisterOperators::options().kernel(c10::TensorTypeId::CPUTensorId,
+                                           [](at::Tensor input, at::Tensor weight, c10::optional<at::Tensor> bias,
+                                              std::vector<int64_t> stride, std::vector<int64_t> padding,
```

Comment:
are we always passing these std::vector<> by value instead of by reference? what's the common practice here?
Tensor is fine as they are just pointers.

Reply:
Good catch. I'm thinking of c10::ArrayRef, but it has compile error "error: no type named 'element_type' in 'c10::ArrayRef<long long>'". @smessmer do you know why it does not work? I'm fixing it using another solution, the boxed op registration.